src/inverse.jl

```diff
@@ -52,6 +52,27 @@ You can check your implementation with [`InverseFunctions.test_inverse`](@ref).
 inverse(f)
 export inverse
 
+
+
+"""
+    struct NoInverse{F}
+
+An instance `NoInverse(f)` signifies that that `inverse(f)` is not
+defined.
+"""
+struct NoInverse{F}
+    f::F
+end
+export NoInverse
```

Comment:
Do we really want to export it? I would have assumed that the average user of InverseFunctions doesn't have to know about it. But on the other hand it makes code such as `inverse(f) isa InverseFunctions.NoInverse` more official...

What about adding an exported
```julia
hasinverse(f) = !isa(inverse(f), NoInverse)
```
? Then one does not have to implement `hasinverse` (which is nice since it keeps the downstream implementations simple) and `inverse(f)` would still work and produce a descriptive error message if called.

Reply:
I would export it `NoInverse` so people can do tests like `inverse(f) isa NoInverse` conveniently. But I would suggest not to add `hasinverse` so that people don't try to specialize/abuse it.

Reply:
OK, I'm fine with it either way. We can always reevaluate.